test/jasmine/tests/sankey_test.js

```diff
@@ -1009,56 +1011,116 @@ describe('sankey tests', function() {
         });
     });
 
-    describe('Test drag interactions:', function() {
-        var gd;
+    describe('Test drag interactions', function() {
+        ['freeform', 'perpendicular'].forEach(function(arrangement) {
```

Comment:
Unfortunately, I am struggling quite a bit to get tests to pass for `snap` (which is the default `arrangement`). It seems like by the time we call `getBoundingClientRect();` in `getNodeCoords`, the DOM hasn't been updated yet and the tests fail. I think we should try to add it here so we :lock: down the dragging behavior.

Reply:
Resolved in 1a89a1e! We now have tests for drag interactions for every arrangement! :tada: